index.bs

```diff
@@ -1284,45 +1219,36 @@ The Module constructor is the initial value of the Module property of the the Re
 
-<h4 id="new-module">Module(descriptors[, executor[, evaluate]])</h4>
+<h4 id="new-module">Module(obj[, evaluate])</h4>
 
-When Module is called with arguments <i>descriptors</i>, <i>executor</i>, and <i>evaluate</i>, the following steps are taken:
+When Module is called with arguments <i>obj</i> and <i>evaluate</i>, the following steps are taken:
 
 <emu-alg>
 1. Let _realm_ be the current Realm Record.
 1. Let _env_ be NewModuleEnvironment(_realm_.[[globalEnv]]).
-1. If Type(_descriptors_) is not Object, throw a *TypeError* exception.
-1. Let _exportDescriptors_ be ParseExportsDescriptors(_descriptors_). // TODO: interleave the subsequent loop with parsing?
+1. If Type(_obj_) is not Object, throw a *TypeError* exception.
+1. Let _objValue_ be ? ToObject(_obj_).
 1. Let _localExports_ be a new empty List.
 1. Let _indirectExports_ be a new empty List.
 1. Let _exportNames_ be a new empty List.
 1. Let _envRec_ be _env_'s environment record.
-1. For each _desc_ in _exportDescriptors_, do:
-  1. Let _exportName_ be _desc_.[[Name]].
-  1. Append _exportName_ to _exportNames_.
-  1. If _desc_ is an Indirect Export Descriptor, then:
-    1. Let _otherMod_ be _desc_.[[Module]].
-    1. Let _resolution_ be ? _otherMod_.ResolveExport(_desc_.[[Import]], « »).
-    1. If _resolution_ is *null*, then throw a *SyntaxError* exception.
-    1. Append the record {[[Key]]: _exportName_, [[Value]]: _resolution_} to _indirectExports_.
-  1. Else,
-    1. Append _exportName_ to _localExports_.
-    1. If _desc_ is an Immutable Export Descriptor, then:
-      1. Perform ? _envRec_.CreateImmutableBinding(_exportName_, *true*).
+1. Let _keys_ be ? _objValue_.[[OwnPropertyKeys]]().
+1. Repeat for each element _exportName_ of _keys_ in List order,
+  1. Let _propDesc_ be ? _objValue_.[[GetOwnProperty]](_exportName_).
+  1. If _propDesc_ is not *undefined* and _propDesc_.[[Enumerable]] is *true*, then:
+    1. Let _exportObj_ be ? Get(_objValue_, _exportName_).
+    1. If _exportObj_ is not an Export Object, throw a *TypeError* exception.
+    1. Append _exportName_ to _exportNames_.
+    1. If _exportObj_.[[resolvedExport]] is not *undefined*, then:
+      1. Let _resolution_ be _exportObj_.[[resolvedExport]].
+      1. Append the record {[[Key]]: _exportName_, [[Value]]: _resolution_} to _indirectExports_.
     1. Else,
-      1. Assert: _desc_ is a Mutable Export Descriptor.
-      1. Perform ? _envRec_.CreateMutableBinding(_exportName_, *false*).
-    1. If _desc_.[[Initialized]] is *true*, then:
-      1. Call _envRec_.InitializeBinding(_exportName_, _desc_.[[Value]]).
-1. If _evaluate_ is not *undefined*, then
-  1. If IsCallable(_evaluate_) is *false*, throw a new *TypeError* exception.
+      1. Append _exportName_ to _localExports_.
+      1. Call _envRec_.CreateIndirectBinding(_exportName_, _exportObj_.[[target]], _exportObj_.[[targetKey]]).
```

Comment:
`CreateIndirectBinding` have to be spec'd in 262, it will be very similar to [CreateImportBinding](https://tc39.github.io/ecma262/#sec-createimportbinding), but it opens the door for Export Objects.